src/debug_utils.cc

```diff
@@ -299,6 +320,106 @@ void CheckedUvLoopClose(uv_loop_t* loop) {
   CHECK(0 && "uv_loop_close() while having open handles");
 }
 
+std::vector<std::string> NativeSymbolDebuggingContext::GetLoadedLibraries() {
+  std::vector<std::string> list(0);
+#ifdef __linux__
+  dl_iterate_phdr(
+      [](struct dl_phdr_info* info, size_t size, void* data) {
+        std::vector<std::string>* list =
+            static_cast<std::vector<std::string>*>(data);
+        if (info->dlpi_name != nullptr && *info->dlpi_name != '\0') {
+          list->push_back(info->dlpi_name);
+        }
+        return 0;
+      },
+      &list);
+#elif __APPLE__
+  int i = 0;
+  for (const char* name = _dyld_get_image_name(i); name != nullptr;
+       name = _dyld_get_image_name(++i)) {
+    list.push_back(name);
+  }
+
+#elif _AIX
+  // We can't tell in advance how large the buffer needs to be.
+  // Retry until we reach too large a size (1Mb).
+  const unsigned int buffer_inc = 4096;
+  unsigned int buffer_size = buffer_inc;
+  MallocedBuffer<char> buffer(buffer_size);
+  int rc = -1;
+  while (rc != 0 && buffer_size < 1024 * 1024) {
+    rc = loadquery(L_GETINFO, buffer.data, buffer_size);
+    if (rc == 0) {
+      break;
+    }
+    buffer_size += buffer_inc;
+    buffer = MallocedBuffer<char>(buffer_size);
+  }
+  if (rc == 0) {
+    char* buf = buffer.data;
+    ld_info* cur_info = nullptr;
+    do {
+      std::ostringstream str;
+      cur_info = reinterpret_cast<ld_info*>(buf);
+      char* member_name = cur_info->ldinfo_filename +
+          strlen(cur_info->ldinfo_filename) + 1;
+      if (*member_name != '\0') {
+        str << cur_info->ldinfo_filename << "(" << member_name << ")";
+        list.push_back(str.str());
+        str.str("");
```

Comment:
This isn't exactly wrong but we overwhelmingly use snprintf() rather than std::ostringstream to format strings.

Reply:
sure; as we are dealing with strings of arbitrary length, `snprintf` would require a static char array and then we might need to worry about its upper bounds etc.? this stream looks very handy to me; unless you have objections. :)